### PR TITLE
More centaur patience and logging.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala [hunk 1]
```diff
@@ -40,7 +41,19 @@ object CentaurCromwellClient {
     .getAs[Map[String, List[String]]]("centaur.metadata-args")
 
   def submit(workflow: Workflow): IO[SubmittedWorkflow] = {
-    sendReceiveFutureCompletion(() => cromwellClient.submit(workflow.toWorkflowSubmission(refreshToken = CentaurConfig.optionalToken)))
+    sendReceiveFutureCompletion(() => {
+      val submitted = cromwellClient.submit(workflow.toWorkflowSubmission(refreshToken = CentaurConfig.optionalToken))
+      submitted.biSemiflatMap(
```
Comment (mcovarr) [inline centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala]: 😮

Comment (kshakir) [reply centaur/src/main/scala/centaur/api/CentaurCromwellClient.scala]: I couldn't tell if it was "bad" to `someIo.map { orig => log.info("Am I in 'correctly' in an IO?"); orig }`, or if the log should be also wrapped in an `IO(log.info(...))` and flatMap-ed. I went with the latter, possibly belt-and-suspenders.

**Comment by cjllanwarne** [comment]
I spoke too soon, seeing a lot of `23:47:24.836 [centaur-acting-like-a-system-akka.actor.default-dispatcher-9] ERROR centaur.api.CentaurCromwellClient$ - Submitting invalid_inputs_json_object returned 400 Bad Request`. 

Maybe only log as an error if the return code is not 4xx?

**Comment by kshakir** [comment]
Cool, I'll switch the `4XX` series to `log.info()`.

**Pull request merged**